sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-keyvault-certificates-client-side/README.md

```diff
@@ -44,15 +39,89 @@ Response from "https://localhost:8443/": Hello World
        }
     }
     ``` 
-1. Start azure-spring-boot-sample-keyvault-certificates-server-side's SampleApplication
-1. Start azure-spring-boot-sample-keyvault-certificates-client-side's SampleApplication with [MTLS] configuration.  
+1. Add properties in application.yml of `server side` on the base of current configuration:
+    ```yaml
+    server:
+      ssl:
+        client-auth:         # Used for MTLS
+        trust-store-type:    # Used for MTLS   
+    ```
+1. Start azure-spring-boot-sample-keyvault-certificates-client-side's SampleApplication by running command:
+   ```
+   mvn spring-boot:run
+   ```
 1. When the [MTLS] server starts, `tls endpoint`(http://localhost:8080/tls) will not be able to access the resource. Access http://localhost:8080/mtls
 
-Then you will get
-```text
-Response from "https://localhost:8443/": Hello World
-```
+    Then you will get
+    ```text
+    Response from "https://localhost:8443/": Hello World
+    ```
 
+### Run sample with managed identity
+1. If you are using managed identity instead of service principal, use below properties in your `application.yml`:
+
+    ```yaml
+    azure:
+      keyvault:
+        uri: ${KEY_VAULT_URI}
+    #    managed-identity: # client-id of the user-assigned managed identity to use. If empty, then system-assigned managed identity will be used.
+    ```
+    Make sure the managed identity can access target Key Vault.
+1. Set environment variables created in `azure-spring-boot-sample-keyvault-certificates-server-side` application by running command:
+   ```
+   source script/setup.sh
+   ```
+
+#### Using TLS with managed identity
+1. Replace the `restTemplateWithTLS` bean in `SampleApplicationConfiguration.java` as
+    ```java
+    @Bean
+    public RestTemplate restTemplateWithTLS() throws Exception {
+        KeyStore trustStore = KeyStore.getInstance("AzureKeyVault");
+        KeyVaultLoadStoreParameter parameter = new KeyVaultLoadStoreParameter(
+            System.getProperty("azure.keyvault.uri"),
+            System.getProperty("azure.keyvault.managed-identity"));
+        trustStore.load(parameter);
+        SSLContext sslContext = SSLContexts.custom()
+                                           .loadTrustMaterial(trustStore, null)
+                                           .build();
+        SSLConnectionSocketFactory socketFactory = new SSLConnectionSocketFactory(sslContext,
+                                                                                  (hostname, session) -> true);
+        CloseableHttpClient httpClient = HttpClients.custom()
+                                                    .setSSLSocketFactory(socketFactory)
+                                                    .build();
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory(httpClient);
+
+        return new RestTemplate(requestFactory);
+    }
```

Comment:
Could you please add reference to `SampleApplicationConfiguration.java`? So we can keep it updated.